solarforecastarbiter/io/reference_observations/reference_data.py

```diff
@@ -0,0 +1,349 @@
+"""A CLI tool for creating reference sites and observations and updating them
+with data from their respective API.
+
+Requires that you set the environment variables:
+
+    SFA_REFERENCE_TOKEN
+        A valid access token for the SolarForecastArbiter reference user.
+    SFA_API_BASE_URL
+        The base url of the SolarForecastArbiter API to use.
+        e.g. https://api.solarforecastarbiter.org
+
+As new networks are added, their name should be added as it appears in
+`sfa_reference_data.csv` to `NETWORK_OPTIONS`. A module should be created
+to handle observation initialization and data update. Each of these network
+specific modules should implement two functions:
+
+    initialize_site_observations(api, site)
+
+        Where api is an `io.api.APISession` and site is a `datamodel.Site`.
+
+    update_observation_data(api, sites, observations, start, end)
+
+        Where api is an `io.api.APISession`, sites is the result of
+        `APISession.list_sites`, observations is the result of
+        `APISession.list_observations``start` and `end` are datetime
+        objects.
+The module should then be imported and added to `NETWORKHANDLER_MAP` below,
+so that it may be selected based on command line arguments. See the existing
+mappings for an example.
+"""
+import argparse
+import inspect
+import json
+import logging
+import os
+import sys
+
+
+import pandas as pd
+from requests.exceptions import HTTPError
+
+
+from solarforecastarbiter.datamodel import Site
+from solarforecastarbiter.io.api import APISession
+from solarforecastarbiter.io.reference_observations import (
+    surfrad,
+    solrad,
+    crn,
+    midc,
+    srml,
+    sandia,
+    common
+)
+
+
+# maps network names to the modules that interact with their api
+NETWORKHANDLER_MAP = {
+    'NOAA SURFRAD': surfrad,
+    'NOAA SOLRAD': solrad,
+    'NOAA USCRN': crn,
+    'UO SRML': srml,
+    'NREL MIDC': midc,
+    'SANDIA': sandia,
+}
+
+# list of options for the 'network' argument
+NETWORK_OPTIONS = ['NOAA SURFRAD', 'NOAA SOLRAD', 'NOAA USCRN', 'NREL MIDC',
+                   'UO SRML', 'ARM', 'SANDIA']
+
+SCRIPT_DIR = os.path.dirname(
+    os.path.abspath(inspect.getfile(inspect.currentframe())))
+DEFAULT_SITEFILE = os.path.join(SCRIPT_DIR, "sfa_reference_sites.csv")
```

Comment:
I think the way to do this is to use ``resource_filename``from setuptools https://setuptools.readthedocs.io/en/latest/pkg_resources.html#resource-extraction

so 
```python
from pkg_resources import resource_filename, Requirement
...
DEFAULT_SITEFILE = resource_filename(Requirement.parse('solarforecastarbiter'), 'solarforecastarbiter/io/reference_observations/sfa_reference_sites.csv')
```